### PR TITLE
[Dashboard] Total in out card enhancement

### LINKED ISSUE #2853: [Dashboard] Total on and out card - Payments subsection
### Story

- As a USER, I want to be able to take quick actions such as pay, add funds, claim funds, create a team.

### Description

Given the focus on financial actions and tools, we want to display the out funds/payments done in the last month for the selected team/entire colony.
Include the Pay button that opens the side action panel with the simple payment/ payment builder action type is pre-selected and the current team as the team

## Implementation
Relies on the queries for the monthly chart

### This issue is complete when...

- [ ] Dashboard page includes a 'Total in and out' card displaying the last month's total out payments for the selected team/entire colony
- [ ] Dashboard page includes a 'Total in and out' card displaying the Pay button in the Payments subsection

[Figma link](https://www.figma.com/design/gac6xVgGNCAge5Bia933dp/Dashboard?node-id=4859-36367&t=3YmH0OCdhMXflaYC-0)
[Figma link empty state](https://www.figma.com/design/gac6xVgGNCAge5Bia933dp/Dashboard?node-id=4867-40639&t=3YmH0OCdhMXflaYC-0)

**Feature branch**: `feat/enhanced-dashboard`

### PR DESCRIPTION
**Pull request by mmioana**
## Description

- This PR addresses the `Payments` and `Income` sections from the `Total in and out` card

Screenshot 2024-09-10 at 10 51 10

## Testing

TODO: Please check the payments and income sections are according to [Figma](https://www.figma.com/design/gac6xVgGNCAge5Bia933dp/Dashboard?node-id=4878-89693&node-type=frame&t=xND2OvegDeZWkrKJ-0)

* Step 1. Run `create-data` script
* Step 2. Go to `planex` dashboard page http://localhost:9091/planex/
* Step 3. Check the Payments and Income values are the same as the ones displayed in the chart for the current month
* Step 4. Create a `Mint tokens` action/motion
* Step 5. After a bit, the income value should update
* Step 6. In order to not directly claim the minted tokens, please create a patch file with the following content - you can name it `.diff`
```
diff --git a/src/redux/sagas/actions/mintTokens.ts b/src/redux/sagas/actions/mintTokens.ts
index d13052bee..12c7658e7 100644
--- a/src/redux/sagas/actions/mintTokens.ts
+++ b/src/redux/sagas/actions/mintTokens.ts
@@ -62,18 +62,18 @@ function* createMintTokensAction({
       ready: false,
     });
 
-    yield fork(createTransaction, claimColonyFunds.id, {
-      context: ClientType.ColonyClient,
-      methodName: 'claimColonyFunds',
-      identifier: colonyAddress,
-      params: [nativeTokenAddress],
-      group: {
-        key: batchKey,
-        id: metaId,
-        index: 1,
-      },
-      ready: false,
-    });
+    // yield fork(createTransaction, claimColonyFunds.id, {
+    //   context: ClientType.ColonyClient,
+    //   methodName: 'claimColonyFunds',
+    //   identifier: colonyAddress,
+    //   params: [nativeTokenAddress],
+    //   group: {
+    //     key: batchKey,
+    //     id: metaId,
+    //     index: 1,
+    //   },
+    //   ready: false,
+    // });
 
     if (annotationMessage) {
       yield fork(createTransaction, annotateMintTokens.id, {
@@ -91,7 +91,7 @@ function* createMintTokensAction({
     }
 
     yield takeFrom(mintTokens.channel, ActionTypes.TRANSACTION_CREATED);
-    yield takeFrom(claimColonyFunds.channel, ActionTypes.TRANSACTION_CREATED);
+    // yield takeFrom(claimColonyFunds.channel, ActionTypes.TRANSACTION_CREATED);
 
     if (annotationMessage) {
       yield takeFrom(
@@ -108,9 +108,9 @@ function* createMintTokensAction({
       },
     } = yield waitForTxResult(mintTokens.channel);
 
-    yield initiateTransaction(claimColonyFunds.id);
+    // yield initiateTransaction(claimColonyFunds.id);
 
-    yield waitForTxResult(claimColonyFunds.channel);
+    // yield waitForTxResult(claimColonyFunds.channel);
 
     yield createActionMetadataInDB(txHash, customActionTitle);
 

```
* Step 7. Apply the patch running `git apply <patch_filename>`
* Step 8. Create another `Mint tokens` action/motion
* Step 9. Check the Claim button appears in the `Income` section

Screenshot 2024-09-10 at 11 13 24

* Step 10. Claim the pending claimable funds and check the `Funds claimed` button appears

Screenshot 2024-09-10 at 11 14 59

* Step 11. Click the `Add` button and check the Add funds modal appears

Screenshot 2024-09-10 at 11 16 41

* Step 12. Click the `Pay` button and check the `Simple payment` action sidebar is opened with the current team pre-filled

## Diffs

**New stuff** ✨

* The `Payments` and `Income` sections for the `Total in and out` card

**Deletions** ⚰️

* `BalanceModal` was discarded in favour of  `AddFundsModal`

Resolves #2853 #2854 #2855 #2856

### CODE REVIEW

**Review by Nortsova — APPROVED** [review]
Values are the same for the chart and title: 
image

After minting the token and refreshing, the income value is updated: 
image

After the patch and minting the token, I can see the claim button:
image

Funds successfully claimed:
Sep-10-2024 21-02-58

Clicking the add button for funds: 
image

Note: Shouldn't everything be blurred? Right now, it looks like only the header is blurred.

Also, after paying someone, the "Payments" value hasn't changed.

File: src/utils/teams.ts [hunk 1]
```diff
@@ -78,6 +78,45 @@ export const getTeamHexColor = (color?: DomainColor) => {
   }
 };
 
+export const getTeamHexSecondaryColor = (color?: DomainColor) => {
```
Comment (rdig) [inline src/utils/teams.ts:81]: :+1:

**Review by rdig — CHANGES_REQUESTED** [review]
Mostly good, well... all good, except a bit of UI nonsense that happens with larger amounts, that's the only reason I've rejected this.

Loading:

Screenshot from 2024-09-12 09-23-54

Initial:

Screenshot from 2024-09-12 09-23-57

Mint with claim:

Screenshot from 2024-09-12 09-25-57

Updates after mint:

Screenshot from 2024-09-12 09-26-08

Mint without claim:

Screenshot from 2024-09-12 09-33-39

Claiming:

Screenshot from 2024-09-12 09-33-59
Screenshot from 2024-09-12 09-34-25

Deposit Modal:

Screenshot from 2024-09-12 09-34-51

Simple Payment pre-filled:

Screenshot from 2024-09-12 09-35-19

As you can see when "Funds claimed" success message shows up, if the amount is large enoung _(well... long enough anyway)_, it will push the button(s) the next row, causing the UI to break slightly.

Screenshot from 2024-09-12 09-34-25

As a side note, and not related to this PR necessarily, I couldn't manage to apply your patch manually:

Screenshot from 2024-09-12 09-28-53

Which is super weird, since I made the changes myself, by hand, and then compared your patch and my changes, and, besides one empty new line, they're the same.

The empty line can't really be the cause since that is removed by default automatically anyway when you save the file.

Screenshot from 2024-09-12 09-32-33

**Comment by iamsamgibbs** [comment]
> Regarding the modal background blur, I didn't change any behaviour regarding it, so if @melyndav considers this to be needed, we can create a separate issue for it 🌞

@mmioana Just a note to say this is blurred on master so something has broken along the way (If the issue hasn't been introduced in this PR, then I suspect this is a problem on the navigation branch, but I haven't checked it there, it probably makes more sense to fix it there if that is the case)
 
Screenshot 2024-09-12 at 09 10 59

**Comment by mmioana** [comment]
> Mostly good, well... all good, except a bit of UI nonsense that happens with larger amounts, that's the only reason I've rejected this.
> 
> Loading:
> 
> Screenshot from 2024-09-12 09-23-54
> 
> Initial:
> 
> Screenshot from 2024-09-12 09-23-57
> 
> Mint with claim:
> 
> Screenshot from 2024-09-12 09-25-57
> 
> Updates after mint:
> 
> Screenshot from 2024-09-12 09-26-08
> 
> Mint without claim:
> 
> Screenshot from 2024-09-12 09-33-39
> 
> Claiming:
> 
> Screenshot from 2024-09-12 09-33-59 Screenshot from 2024-09-12 09-34-25
> 
> Deposit Modal:
> 
> Screenshot from 2024-09-12 09-34-51
> 
> Simple Payment pre-filled:
> 
> Screenshot from 2024-09-12 09-35-19
> 
> As you can see when "Funds claimed" success message shows up, if the amount is large enoung _(well... long enough anyway)_, it will push the button(s) the next row, causing the UI to break slightly.
> 
> Screenshot from 2024-09-12 09-34-25
> 
> As a side note, and not related to this PR necessarily, I couldn't manage to apply your patch manually:
> 
> Screenshot from 2024-09-12 09-28-53
> 
> Which is super weird, since I made the changes myself, by hand, and then compared your patch and my changes, and, besides one empty new line, they're the same.
> 
> The empty line can't really be the cause since that is removed by default automatically anyway when you save the file.
> 
> Screenshot from 2024-09-12 09-32-33

Thank you for reviewing this @rdig 🙏 

I'll take care of the buttons breaking into the new line 💯 

Regarding the patch this is very weird. I assume it might have to do with the new line, but I'll try again later today

**Comment by mmioana** [comment]
> > Regarding the modal background blur, I didn't change any behaviour regarding it, so if @melyndav considers this to be needed, we can create a separate issue for it 🌞
> 
> @mmioana Just a note to say this is blurred on master so something has broken along the way ~(If the issue hasn't been introduced in this PR, then I suspect this is a problem on the navigation branch, but I haven't checked it there, it probably makes more sense to fix it there if that is the case)~ Confirming this is an issue on `ui/new-navigation-ui` and it should be fixed there. I'll open an issue for it.
> 
> Screenshot 2024-09-12 at 09 10 59

Thank you @iamsamgibbs for checking this 🙏 💯

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Nice work on this! I wasn't able to apply the diff either, but was otherwise able to follow the testing steps.

Income showing the correct value:

Screenshot 2024-09-13 at 12 11 38

Then after minting:

Screenshot 2024-09-13 at 12 13 39

Then after minting (without the auto claim)

Screenshot 2024-09-13 at 12 19 24

The funds claim button appeared correctly, but the value didn't refresh:

Screenshot 2024-09-13 at 12 19 42

Only after a refresh did it update:

Screenshot 2024-09-13 at 12 21 50

This video shows it a little better:

https://github.com/user-attachments/assets/bf27dff8-628c-44ba-bd5b-abff66e6aa82

The modal and action form open correctly:
Screenshot 2024-09-13 at 12 29 43

Screenshot 2024-09-13 at 12 29 36

I couldn't get any payments to show when the "All teams" filter was selected. (Note the payments to a member and another colony in the activity feed)

Screenshot 2024-09-13 at 12 35 24

Payments do show when "General" or any other team is selected though.

Screenshot 2024-09-13 at 13 25 04

Payments appear to be a bit broken however, in this video I make a payment of 1,000 CREDS, but almost £2,000 is shown.

https://github.com/user-attachments/assets/30a38ec8-44f7-4582-844a-6d5e6390c348

I also noticed some differences in the loading skeleton:

Screenshot 2024-09-13 at 12 57 15

Figma:
Screenshot 2024-09-13 at 12 57 52

I'm guessing the missing block is the trend arrows which are yet to be added - but some of the loaders appear more rounded in figma.

Appreciate some of this is likely out of scope for this issue! 😅

**Comment by mmioana** [comment]
@rdig and @iamsamgibbs thank you a lot for your throughout review 🙏 

After discussing with @melyndav and @arrenv, the button shown after claiming funds will only display `Claimed` - without the tick icon - thus the buttons overflowing to the next row shouldn't happen anymore

Also, I have updated the patch file content - there might have been some spacing issues causing the `git apply <_filename_here_>` to not work as expected

@iamsamgibbs I have added a refetch call after claiming the funds. Thank you for spotting this 💯 
Regarding the payments value, there is another issue I have created after talking with @jakubcolony as there definitely is an overlap between the simple payment actions and expenditures, thus having a wrong value returned from the lambda - https://github.com/JoinColony/colonyCDapp/issues/3067

Also, I have increased the border-radius for the button's loading skeletons from `4px` to `8px`, as per Figma

Thank you for spotting these issues and please give it another go 🙏

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Thanks for making those changes! Tested and everything is still working as expected. The diff worked this time too thanks!

Rounded corners look better on the loading skeleton:

Screenshot 2024-09-16 at 15 54 29

But the refetch doesn't seem to be happening after a claim, not sure if this is just me?

https://github.com/user-attachments/assets/7ad5f749-9b0f-4b8a-b751-098be08bf6b5

**Comment by mmioana** [comment]
Thank you @iamsamgibbs for checking again! 
I have pushed now a different fix, so that both the Income/Payments values get updated and the bar chart, if you want to give it another go 🙏

**Review by iamsamgibbs — APPROVED** [review]
Nice job! The income is refetched properly now when claiming. All looks good to me!

https://github.com/user-attachments/assets/a36b3f3b-8f4c-431c-9a0a-f35ff6c67e4e

**Review by rdig — APPROVED** [review]
Re-approved

**Review by Nortsova — APPROVED** [review]
Thanks for changes! 🙌  Reapproved ✅

Value was updated after minting tokens:
image
I was able to claim incoming funds:
claim
After that claim button disappear

**Pull request merged**